### ISSUE TITLE
asynchronisation "completed" and PhedexInjector

### ISSUE THREAD

**Issue by vlimant** [dmwm/WMCore#8148]
looking at 
https://cmsweb.cern.ch/reqmgr2/fetch?rid=zhenhu_RVCMSSW_9_2_11Higgs200ChargedTaus_13_170907_065054_3794

completed: 2017-09-07 19:23:15 UTC

and

2017-09-07 21:31:24,252:140362548885248:INFO:PhEDExInjectorPoller:About to inject 2 files for block /RelValHiggs200ChargedTaus_13/CMSSW_9_2_11-TkAlMuonIsolated-92X_upgrade2017_realistic_v11-v1/ALCARECO#de69f5a6-9402-11e7-bff4-02163e00d7b3
2017-09-07 21:31:36,417:140362548885248:INFO:PhEDExInjectorPoller:Block closed in PhEDEx: /RelValHiggs200ChargedTaus_13/CMSSW_9_2_11-TkAlMuonIsolated-92X_upgrade2017_realistic_v11-v1/ALCARECO#de69f5a6-9402-11e7-bff4-02163e00d7b3

it looks like files were injected in phedex *after* the workflow went into completed.
This has probably been discussed in the past, but either we get a new status (injecting ? completing?), or we synchronize "completed" with "no files left to be injected in DBS or phedex".

Main reason is, how/when does Ops know when to go look for actually missing files, if there could be a silent aysnchronisation. Waiting is not an option operation-wise

**Comment by ticoann** [comment]
I thought Unified checks this. Anyway, we will try to add this feature by next release

**Comment by vlimant** [comment]
yes, unifies check for phedex/dbs inconsistency, and sometimes there is something that needs to be taken care of with transfer team. The point is that if there can systematically be such inconsistency, one has to wait n (= how much?) hours before checking request in "completed" status before checking and acting on the inconsistency. In short, there is no way to know if it's just a delay or just files missing/invalidated in the wild

**Comment by vlimant** [comment]
https://its.cern.ch/jira/browse/CMSCOMPPR-1361 for a use-case where having the synchronisation is mandatory to make sense of the "completed" status

**Comment by bbockelm** [comment]
@vlimant - is this still high-priority?  It has lingered for an awfully long time.

**Comment by amaltaro** [comment]
I'll dump whatever I have to do in October and get this one fixed. Or I close it and we say we can't fix it and we need to live with this forever.

**Comment by vlimant** [comment]
which is it going to be ?

**Comment by ticoann** [comment]
Sorry this was my ticket. I will take care of this.

**Comment by amaltaro** [comment]
Linked issue #9543 here, which is reporting the same problem of workflows sitting in `completed` status for a long time (hours? days? weeks?) while still waiting data to get injected on DBS (or on DBS and Phedex).

**Comment by vlimant** [comment]
#9543 is about an additional issue on dbsuploader that has value in being fixed on its own, rather than trying to get a synchronization (longer term) of the status with files injections.
If you do not fix #9543 now, workflows will stay for long time in limbo before getting to completed, and things will look bad all the same overall.

**Comment by klannon** [comment]
@amaltaro I'm impressed that this issue from 2017 (!) appears in our workplan.  Given that this refers to PhEDEx and might include references to other outdated concepts, perhaps you could spend 60 seconds writing a brief updated description at the top (with a new title too, perhaps?) so that a modern audience appreciates the intentions here?

**Comment by amaltaro** [comment]
@klannon you are right, apologies for not getting earlier to this. I have just refactored the original issue description.

I wanted to note though that the P&R team does not consider this issue important for Q4, they are actually interested in https://github.com/dmwm/WMCore/issues/11729 (and of course, to no longer have file mismatch in WM, which is a very generic problem). Said that, I am removing it from the 2023/Q4 board.

**Comment by klannon** [comment]
Fair enough.  Thanks!
